src/components/Dropdown/tests/Dropdown.test.tsx

```diff
@@ -26,45 +43,60 @@ describe(`Dropdown`, () => {
   });
 
-  it(`Must render 1 dropdown at a time when clicked`, async () => {
-    renderWithRedux(<Dropdown {...mockProps} categories={categoriesMock} />);
-
-    const firstCategory = screen.getByRole(`heading`, { name: /reuniao/i });
-    const secondCategory = screen.getByRole(`heading`, { name: /comida/i });
-
-    await userEvent.click(firstCategory);
+  it(`When the component is rendered, there should be a category dropdown button`, async () => {
+    renderWithRedux(<Dropdown {...mockProps} />);
 
-    const firstCategoryProduct = await screen.findAllByText(/pão/i);
+    const categoryDropdownButton = screen.getByRole(`heading`, {
+      name: /reuniao/i,
+    });
 
-    expect(firstCategoryProduct).toHaveLength(1);
+    expect(categoryDropdownButton).toBeInTheDocument();
 
-    await userEvent.click(secondCategory);
+    const firstCategoryProduct = screen.queryByText(/pão/i);
+    const secondCategoryProduct = screen.queryByText(/guaraná antartica/i);
+    const thirdCategoryProduct = screen.queryByText(/guaraná jesus/i);
+    const fourthCategoryProduct = screen.queryByText(/smash burguer/i);
 
-    const secondCategoryProduct = await screen.findAllByText(/pão/i);
-
-    expect(secondCategoryProduct).toHaveLength(1);
+    expect(firstCategoryProduct).not.toBeInTheDocument();
+    expect(secondCategoryProduct).not.toBeInTheDocument();
+    expect(thirdCategoryProduct).not.toBeInTheDocument();
+    expect(fourthCategoryProduct).not.toBeInTheDocument();
   });
 
-  it(`Only shows the product that was searched for`, async () => {
-    renderWithRedux(<Dropdown {...mockProps} categories={categoriesMock} />);
+  it(`When the dropdown is clicked, there must be products inside it`, async () => {
+    renderWithRedux(<Dropdown {...mockProps} />);
 
-    const firstCategory = screen.getByRole(`heading`, { name: /reuniao/i });
+    const categoryDropdownButton = screen.getByRole(`heading`, {
+      name: /reuniao/i,
+    });
 
-    await userEvent.click(firstCategory);
+    let firstCategoryProduct = screen.queryByText(/pão/i);
+    let secondCategoryProduct = screen.queryByText(/guaraná antartica/i);
+    let thirdCategoryProduct = screen.queryByText(/guaraná jesus/i);
+    let fourthCategoryProduct = screen.queryByText(/smash burguer/i);
 
-    const searchInput = screen.getAllByPlaceholderText(
-      `Pesquisar produto por nome...`
-    )[0];
+    expect(firstCategoryProduct).not.toBeInTheDocument();
+    expect(secondCategoryProduct).not.toBeInTheDocument();
+    expect(thirdCategoryProduct).not.toBeInTheDocument();
+    expect(fourthCategoryProduct).not.toBeInTheDocument();
 
-    expect(searchInput).toBeInTheDocument();
+    await userEvent.click(categoryDropdownButton);
 
-    const firstCategoryProduct = await screen.findByText(/pão/i);
+    const modifiedMockProps = {
+      ...mockProps,
+      showDropdown: 0,
+    };
 
-    await userEvent.type(searchInput, `smash burguer`);
+    renderWithRedux(<Dropdown {...modifiedMockProps} />);
```

Comment:
Is this optimal? 2 renders in one test, this should be 2 separated tests, no?

Reply:
But if you are rerendering the component, everything you did before does not aplly to the second render, no? So if you click in 20 things and rerender it after, everything will be back to the initial state so everything before is not valid anymore, so why test it in the same test? Never did 2 renders in a teste but i think that's the way it will work.

Reply:
The modified mock has a prop that’s open the dropdown, before the second render I only verify if the dropdown was open, and in the second I pass the prop that’s open the dropdown, so the dropdown is opened. Not make sense create a second test to this, because the dropdown can init open or close depending the passed prop.

Reply:
But was exactly what you did, you opened the dropdown with the prop and not the click.

![image](https://github.com/trukaSoftware/menu-digital/assets/83774878/32220b13-8119-4ce0-8b09-681d5c2f98d8)


Here I removed the click and the test still passes because you are rerendering the component from 0, the click does nothing, this is a false positive, you are not testing the click here you are only testing the prop.